fdb-extensions/src/main/java/com/apple/foundationdb/async/RangeSet.java

```diff
@@ -270,7 +270,7 @@ private static void checkRange(@Nonnull byte[] begin, @Nonnull byte[] end) {
                     // If we are allowing non-empty ranges, then we just need to fill in the gaps.
                     return AsyncUtil.whileTrue(() -> {
                         byte[] lastSeenBytes = lastSeen.get();
-                        if (afterIterator.onHasNext().isDone() && afterIterator.hasNext()) {
+                        if (MoreAsyncUtil.isCompletedNormally(afterIterator.onHasNext()) && afterIterator.hasNext()) {
```

Comment:
Hmm, I suppose we used to join on the future immediately (via `afterIterator.hasNext()`), so this future would complete exceptionally. Now we'll do the same by returning `afterIterator.onHasNext()`?

Reply:
Yeah, if `afterIterator.onHasNext()` completes exceptionally, then this used to raise an error (with the `hasNext()` call). Now, it propagates the exceptional future to `whileTrue`, which in turn boils the exception back up.

So, in theory, this isn't strictly necessary, but I think *in general* propagating the exceptional future rather than re-raising is better etiquette. I could be wrong though.